.github/workflows/pr-triage.yml

```diff
@@ -0,0 +1,72 @@
+name: PR triage labels
+
+on:
+  pull_request_review:
+    types: [submitted, edited, dismissed]
+  pull_request:
+    types: [synchronize, reopened, ready_for_review, converted_to_draft]
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  label-awaiting-author:
+    name: Manage awaiting-author/awaiting-review labels
+    runs-on: ubuntu-latest
+    steps:
+      - name: Apply/remove labels based on review state or new commits
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const pr = context.payload.pull_request || context.payload.review?.pull_request;
```

Comment:
is it possible to pull this out to a separate file? would be nice to have syntax highlighting and stuff